validation/citrus-validation-json/src/main/java/org/citrusframework/validation/json/JsonElementValidator.java

```diff
@@ -96,21 +96,28 @@ static boolean isIgnoredByPlaceholderOrExpressionList(Collection<String> ignoreE
             return true;
         }
 
-        if (ignoreExpressions.stream().anyMatch(controlEntry::isPathIgnoredBy)) {
-            return true;
-        }
-
-        return false;
+        return ignoreExpressions.stream().anyMatch(controlEntry::isPathIgnoredBy);
     }
 
-
     private void validateJSONArray(JsonElementValidator validator, JsonElementValidatorItem<?> control) {
         var arrayControl = control.ensureType(JSONArray.class);
         if (strict) {
             validateSameSize(control.getJsonPath(), arrayControl.expected, arrayControl.actual);
         }
+        int actualIndex = 0;
         for (int i = 0; i < arrayControl.expected.size(); i++) {
-            if (!isAnyValidItemInActualArray(validator, arrayControl, i)) {
+            if (isIgnoredByPlaceholderOrExpressionList(ignoreExpressions, arrayControl.child(i, i))) {
+                continue;
+            }
+            boolean isValid = false;
+
+            while (!isValid && actualIndex < arrayControl.actual.size()) {
```

Comment:
Without judging whether this approach would make more sense, it is important to notice, that it is different to the previous one. So it is potentially breaking and may result in false positives for tests which are out there. Since there is no need for this "potential improvement", I would simply stay with the previous implementation which only accepted "additional elements" at the end of the array in non strict mode. If you really change the behaviour like this, you need to properly document it and provide a configuration option to switch back to the previous behaviour.

Reply:
As discussed bilateral, we'll keep the "new" behaviour and document it.